### PR TITLE
updated travis and babel versions, as well as dependencies

### PR DESCRIPTION
**Pull request by davidcgong**
These have to be changed everytime yeoman is run to generate a new ship-component. Just make sure 'npm install' works without throwing errors and such, and it should be good?

### CODE REVIEW

File: generators/app/templates/package.json [hunk 1]
```diff
@@ -39,50 +39,53 @@
...
-    "style-loader": "^0.18.2",
-    "webpack": "^3.2.0",
-    "webpack-dev-server": "^2.5.0"
+    "@babel/core": "^7.4.5",
+    "@babel/plugin-transform-runtime": "^7.4.4",
+    "@babel/preset-env": "^7.4.5",
+    "@babel/preset-react": "^7.0.0",
+    "babel-eslint": "^10.0.1",
+    "babel-loader": "^8.0.6",
+    "babel-preset-env": "^1.7.0",
```
Comment (sepandasadi) [inline generators/app/templates/package.json]: This is duplicated. Please go through all the dev and dependencies, and ensure there are not duplicated.

File: generators/app/templates/resource/tasks/config/webpack.js [hunk 2]
```diff
@@ -48,7 +48,6 @@ module.exports = function(grunt) {
           disable: false,
           allChunks: true
         }),
-        new webpack.optimize.UglifyJsPlugin()
```
Comment (sepandasadi) [inline generators/app/templates/resource/tasks/config/webpack.js:51]: try using this syntax
```
const UglifyJsPlugin = require('uglifyjs-webpack-plugin');

config.optimization = {
    minimizer: [
      new UglifyJsPlugin()
    ]
  };

```

File: generators/app/templates/package.json [hunk 3]
```diff
@@ -39,50 +39,53 @@
...
+    "babel-preset-react": "^6.24.1",
+    "babel-runtime": "^6.26.0",
```
Comment (isuttell) [inline generators/app/templates/package.json]: This one should actually be in the dependencies see https://babeljs.io/docs/en/babel-runtime#usage

File: generators/app/templates/resource/tasks/config/webpack.js (hunk 2 above)
Comment (isuttell) [inline generators/app/templates/resource/tasks/config/webpack.js:51]: I don't want to lose any functionality or regress here. So let's add the same functionality back in so we dont have a regression. See https://webpack.js.org/configuration/optimization/

**Review by isuttell — CHANGES_REQUESTED** [review]

File: generators/app/templates/package.json (hunk 1 above)
Comment (davidcgong) [reply generators/app/templates/package.json]: moved non @babel devDependencies to dependencies

**Review by isuttell — APPROVED** [review]
Assuming Sam approves this looks good to me

File: generators/app/templates/package.json [hunk 4]
```diff
@@ -39,50 +39,53 @@
...
-    "prop-types": "^15.5.10",
-    "react-addons-test-utils": "^15.6.0",
-    "react-test-renderer": "^15.6.1",
-    "style-loader": "^0.18.2",
-    "webpack": "^3.2.0",
-    "webpack-dev-server": "^2.5.0"
+    "@babel/core": "^7.4.5",
+    "@babel/plugin-transform-runtime": "^7.4.4",
+    "@babel/preset-env": "^7.4.5",
+    "@babel/preset-react": "^7.0.0",
```
Comment (sepandasadi) [inline generators/app/templates/package.json]: cut and paste this and preset-env into dependencies section

File: generators/app/templates/package.json [hunk 5]
```diff
@@ -39,50 +39,53 @@
   },
   "homepage": "https://github.com/ship-components/generator-ship-components-boilerplate#readme",
   "dependencies": {
-    "babel-runtime": "^6.23.0",
-    "classnames": "^2.2.5",
-    "immutable": "^3.8.1",
-    "react": "^15.6.1",
-    "react-dom": "^15.6.1"
+    "babel-loader": "^8.0.6",
+    "babel-preset-env": "^1.7.0",
```
Comment (sepandasadi) [inline generators/app/templates/package.json]: remove this, and preset-react

**Review by sepandasadi — CHANGES_REQUESTED** [review]

**Review by sepandasadi — APPROVED** [review]

**Pull request merged**